pkg/client/record/event.go

```diff
@@ -145,7 +145,11 @@ func recordEvent(sink EventSink, event *api.Event, updateExistingEvent bool) boo
 	var err error
 	if updateExistingEvent {
 		newEvent, err = sink.Update(event)
-	} else {
+	}
+	// Update can fail because the event may have been removed and it no longer exists.
+	if !updateExistingEvent || (updateExistingEvent && err != nil) {
+		// Making sure that ResourceVersion is empty on creation
+		event.ResourceVersion = ""
```

Comment:
Are you sure you want to do that? I actually found quite nice that no information were lost in this case as well. Even if the event was deleted we still keep the count and first appearance.